app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,37 @@
+class FoodsController < ApplicationController
+  # before_action :set_recipe, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # after implementing devise we have to change User.first.id to current_user.id.
+
+  def index
+    @all_foods = Food.where(user_id: current_user.id)
+  end
+
+  def new
+    @food = Food.new
+  end
+
+  def create
+    @food = Food.new(food_params)
+    @food.user_id = current_user.id
+
+    if @food.save
+      redirect_to user_foods_path, notice: 'Food was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @food = Food.find(params[:id])
+    @food.destroy
+    redirect_to user_foods_path, notice: "#{@food.name} was successfully deleted"
+  end
```

Comment:
- [ ] It will be great to create success and fail paths for your destroy action. If the resource is deleted, do something. If it is not deleted do something else. This will prevent your application from misbehaving incase something unexpected happens

Reply:
@peterrobert not part of the project requirements. you should have marked this as optional.

Reply:
No, this is not optional, when you implement either create destroy or update in your application, is a requirement for us to check if you have done it right. And the only way to do it right is to ensure you have created both success and fail paths. @ram1117